Bottom: 797792d6046a168ae73112f6c49bc2c1022ae255
Top:    30eb81b117fc31110005cd35b5198141eea9e3bb
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-04-13 10:53:58 -0500

Add buildbot role


---
diff --git a/roles/docker-buildbot/README.md b/roles/docker-buildbot/README.md
new file mode 100644
index 0000000..225dd44
--- /dev/null
+++ b/roles/docker-buildbot/README.md
@@ -0,0 +1,38 @@
+Role Name
+=========
+
+A brief description of the role goes here.
+
+Requirements
+------------
+
+Any pre-requisites that may not be covered by Ansible itself or the role should be mentioned here. For instance, if the role uses the EC2 module, it may be a good idea to mention in this section that the boto package is required.
+
+Role Variables
+--------------
+
+A description of the settable variables for this role should go here, including any variables that are in defaults/main.yml, vars/main.yml, and any variables that can/should be set via parameters to the role. Any variables that are read from other roles and/or the global scope (ie. hostvars, group vars, etc.) should be mentioned here as well.
+
+Dependencies
+------------
+
+A list of other roles hosted on Galaxy should go here, plus any details in regards to parameters that may need to be set for other roles, or variables that are used from other roles.
+
+Example Playbook
+----------------
+
+Including an example of how to use your role (for instance, with variables passed in as parameters) is always nice for users too:
+
+    - hosts: servers
+      roles:
+         - { role: username.rolename, x: 42 }
+
+License
+-------
+
+BSD
+
+Author Information
+------------------
+
+An optional section for the role authors to include contact information, or a website (HTML is not allowed).
diff --git a/roles/docker-buildbot/defaults/main.yml b/roles/docker-buildbot/defaults/main.yml
new file mode 100644
index 0000000..9781ada
--- /dev/null
+++ b/roles/docker-buildbot/defaults/main.yml
@@ -0,0 +1,20 @@
+---
+# defaults file for buildbot
+
+# The hostname passed as an envirnment variable into the container
+container_addr_map:
+  buildbot:
+    ip_addr: '127.0.0.1'
+    hostname: buildbot.example.com
+
+# the port mappings used when starting the container
+container_port_map:
+  buildbot:
+    port_args: 
+      - '{{ container_addr_map.buildbot.ip_addr }}:25:25'
+      - '{{ container_addr_map.buildbot.ip_addr }}:465:465'
+      - '{{ container_addr_map.buildbot.ip_addr }}:587:587'
+
+# Network names
+docker_network_backend: 'backend'
+docker_network_frontend: 'frontend'
diff --git a/roles/docker-buildbot/meta/main.yml b/roles/docker-buildbot/meta/main.yml
new file mode 100644
index 0000000..5999cdd
--- /dev/null
+++ b/roles/docker-buildbot/meta/main.yml
@@ -0,0 +1,4 @@
+---
+dependencies: []
+  # List your role dependencies here, one per line. Be sure to remove the '[]' above,
+  # if you add dependencies to this list.
diff --git a/roles/docker-buildbot/tasks/main.yml b/roles/docker-buildbot/tasks/main.yml
new file mode 100644
index 0000000..58c0cfa
--- /dev/null
+++ b/roles/docker-buildbot/tasks/main.yml
@@ -0,0 +1,18 @@
+---
+# tasks file for buildbot
+
+# This will generate passwords for these accounts.
+- assert:
+    that:
+    - buildbot_admin_password is defined
+
+- name: Create buildbot data volume
+  docker_volume:
+    name: '{{ buildbot_dv_name }}'
+
+- name: Start buildbot container
+  docker_container:
+    name: '{{ buildbot_container_name }}'
+    image: '{{ buildbot_image_name }}:{{ docker_image_tag }}'
+    volumes:
+      - '{{ buildbot_dv_name }}:/buildbot-data:z'
diff --git a/roles/docker-buildbot/vars/main.yml b/roles/docker-buildbot/vars/main.yml
new file mode 100644
index 0000000..62beac8
--- /dev/null
+++ b/roles/docker-buildbot/vars/main.yml
@@ -0,0 +1,16 @@
+---
+# vars file for buildbot
+
+# the name of the image being duilt and used for the container
+buildbot_image_repo: https://github.com/novatechweb/docker-buildbot.git
+buildbot_image_name: '{{ docker_registry_username }}/buildbot'
+
+# the name of the container being started
+buildbot_container_name: '{{ docker_name_prefix }}buildbot'
+
+# the name of the data-volume used by the container
+buildbot_dv_name: '{{ docker_name_prefix }}buildbot_DV'
+
+# restore directories to temporarly store data being restored into docker containers
+buildbot_docker_backup_dir: '{{ docker_backup_dir }}/buildbot'
+host_buildbot_docker_restore_dir: '{{ bacula_dest }}{{ buildbot_docker_backup_dir }}'
